tools/crm_mon.c

```diff
@@ -3248,6 +3249,61 @@ print_failed_stonith_actions(FILE *stream, stonith_history_t *history)
     }
 }
 
+/*!
+ * \internal
+ * \brief Print pending stonith actions
+ *
+ * \param[in] stream     File stream to display output to
+ * \param[in] history    List of stonith actions
+ *
+ */
+static void
+print_stonith_pending(FILE *stream, stonith_history_t *history)
+{
+    /* xml-output always shows the full history
+     * so we'll never have to show pending-actions
+     * separately
+     */
+    if (history && (history->state != st_failed) &&
+        (history->state != st_done)) {
+        stonith_history_t *hp;
+
+        /* Print section heading */
+        switch (output_format) {
+            case mon_output_plain:
+            case mon_output_console:
+                print_as("\nPending Fencing Actions:\n");
+                break;
+
+            case mon_output_html:
+            case mon_output_cgi:
+                fprintf(stream, " <hr />\n <h2>Pending Fencing Actions</h2>\n <ul>\n");
+                break;
+
+            default:
+                break;
+        }
+
+        for (hp = history; hp; hp = hp->next) {
+            if ((hp->state == st_failed) || (hp->state == st_done)) {
+                break;
+            }
+            print_stonith_action(stream, hp);
+        }
```

Comment:
Shouldn't this just be if (hp->state == st_exec) { print }? The above looks like it wouldn't print any pending actions after the first failed/done action in the history (unless pending are guaranteed to be first?). Are st_query/st_duplicate states excluded from history?

Reply:
The st_failed, st_done otherwise pending logic is stolen from other places - but might be questionable maybe ...
The list is sorted first for time-stamps and the pending-options don't have one so they are moved front.